pkg/handlers/cni/calico/manifests/docker/installation.yaml

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 D2iQ, Inc. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This section includes base Calico installation configuration.
+# For more information, see: https://docs.projectcalico.org/reference/installation/api
+apiVersion: operator.tigera.io/v1
+kind: Installation
```

Comment:
Having this in the codebase is a great start, but I do think we'll need add support for user overrides here sooner than later. Its a good strategy though to only allow overrides to a limited config, and always provide defaults: Calico `Installation`, CSI `StorageClass`, CPI `ConfigMap/Secret` that may be it really.

Reply:
I think I'd like to move this from being embedded to reference default configmap from the API, and then copying it that way to where it's needed. A default would be created when installing the helm chart, but could be disabled via some value. This would allow users to create their own default configmaps/secrets.

Then allowing configuration per cluster via annotations to use other specific configmaps.

Reply:
I've added the ability to override pod subnet via annotation in latest commit to show how that is possible.

Reply:
>I've added the ability to override pod subnet via annotation in latest commit to show how that is possible.

I think this value should come from the Cluster object directly, idk the side-effects of changing it only 1 place but it would be confusing if those values were different.
https://github.com/kubernetes-sigs/cluster-api/blob/82eff49867008365d3b26f82b55ff29a67880aa7/api/v1beta1/cluster_types.go#L362C15-L362C15
WDYT?

Reply:
Ooo yes I'd missed that. But what would multiple cidrs specified here mean?

Reply:
We could use the equivalent of `calico node status` against a cluster using the Calico API via a controller on a regular poll perhaps.